install/windows/build-image.py

```diff
@@ -18,28 +18,26 @@ def main():
     )
 
     print("Saving container image")
-    subprocess.run(
+    cmd = subprocess.Popen(
         [
             "docker",
             "save",
             "dangerzone.rocks/dangerzone",
-            "-o",
-            "share/container.tar",
-        ]
+        ],
+        stdout=subprocess.PIPE,
     )
 
     print("Compressing container image")
-    chunk_size = 1024
-    with open("share/container.tar", "rb") as f:
-        with gzip.open("share/container.tar.gz", "wb") as gzip_f:
-            while True:
-                chunk = f.read(chunk_size)
-                if len(chunk) > 0:
-                    gzip_f.write(chunk)
-                else:
-                    break
-
-    os.remove("share/container.tar")
+    chunk_size = 4 << 12
+    with gzip.open("share/container.tar.gz", "wb") as gzip_f:
+        while True:
+            chunk = cmd.stdout.read(chunk_size)
+            if len(chunk) > 0:
+                gzip_f.write(chunk)
+            else:
+                break
+
+    cmd.wait(5)
```

Comment:
`Popen.communicate()` serves a different purpose, but I'll try to demystify this bit:

* `Popen.communicate()` basically writes to the process' **stdin** once, consumes the process' **stdout** / **stderr**, and then returns it to the program once it completes.
* In this snippet, we need to constantly read from the process's **stdout**, gzip it, and write to a file, which is different.

So, we can't use `Popen.communicate()` here, and we have to do pipe mangling on our own. Now, why do the Python docs suggest using `Popen.communicate()` in the first place? Because the use case of "write to stdin, read from stdout/stderr" is a very common need, that is surprisingly easy to get it wrong (see https://github.com/freedomofpress/dangerzone/pull/289#discussion_r1060046132 for more details).

Reply:
It seems I have to learn more about pipes in python! And thanks for the clarification!

I wish there was a construction like `with (result.pipe) as stdout:`... or something more pythonic.